### PR TITLE
[rush] Add ability to silence noops in the logs

### PR DESCRIPTION
**Pull request by dmichon-msft**
## Summary
Adds the ability to silence logging from phases that aren't defined in a given project.

## Details
Adds a new configuration option `missingScriptBehavior` to phase definitions in `command-line.json`.
Current allowed values:
`error` - Throws an error if not defined, as `ignoreMissingScript: false` (which it replaces) (default)
`log` - Logs a no-op if not defined, as `ignoreMissingScript: true` (which it replaces)
`silent` - Does not log at all if not defined. Net new behavior.

## How it was tested
Enabled in the local rushstack repository for `_phase:test` and validated that `rush test -v` no longer logged any no-ops for `_phase:test`.

## Impacted documentation
Docs around command-line.json regarding phase definitions.

### CODE REVIEW

File: libraries/rush-lib/src/api/CommandLineConfiguration.ts [hunk 1]
```diff
@@ -25,6 +25,12 @@ export interface IShellCommandTokenContext {
   packageFolder: string;
 }
 
+/**
+ * The set of valid behaviors for a missing script in a project's package.json scripts for a given phase.
+ * @alpha
+ */
+export type IPhaseBehaviorForMissingScript = 'silent' | 'log' | 'error';
```
Comment (iclanton) [inline libraries/rush-lib/src/api/CommandLineConfiguration.ts:32]:
```suggestion
export type PhaseBehaviorForMissingScript = 'silent' | 'log' | 'error';
```

File: common/changes/@microsoft/rush/silent-noops_2023-05-11-22-50.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Add a property \"missingScriptBehavior\" to phase definitions tha can be used to silence missing scripts to reduce log noise.",
```
Comment (octogonz) [inline common/changes/@microsoft/rush/silent-noops_2023-05-11-22-50.json:5]:
```suggestion
      "comment": "Add a property \"missingScriptBehavior\" to phase definitions that can be used to silence missing scripts to reduce log noise.",
```

**Review by octogonz — APPROVED** [review]

File: common/changes/@microsoft/rush/silent-noops_2023-05-11-22-50.json (hunk 2 above)
Comment (iclanton) [inline common/changes/@microsoft/rush/silent-noops_2023-05-11-22-50.json:5]:
```suggestion
      "comment": "(BREAKING API CHANGE) Add a property `missingScriptBehavior` to phase definitions that can be used to silence missing scripts to reduce log noise. This replaces the `ignoreMissingScript` property in the API. The `ignoreMissingScript` property is still supported in the `common/config/rush/command-line.json` config file.",
```

**Comment by Faithfinder** [comment]
Seems like the next logical step would be to silence "Defined no work" for projects that have empty scripts?

**Comment by dmichon-msft** [comment]
> Seems like the next logical step would be to silence "Defined no work" for projects that have empty scripts?

That's precisely what this change does, if you specify `missingScriptBehavior: 'silent'`.

**Comment by Faithfinder** [comment]
> > Seems like the next logical step would be to silence "Defined no work" for projects that have empty scripts?
> 
> That's precisely what this change does, if you specify `missingScriptBehavior: 'silent'`.

Aren't empty scripts different from missing?

**Comment by dmichon-msft** [comment]
> > > Seems like the next logical step would be to silence "Defined no work" for projects that have empty scripts?
> > 
> > 
> > That's precisely what this change does, if you specify `missingScriptBehavior: 'silent'`.
> 
> Aren't empty scripts different from missing?

With `silent`, there is no distinction. With `error`, missing scripts throw an error, while empty scripts report a noop.

**Review by iclanton — APPROVED** [review]

**Review by iclanton — APPROVED** [review]

**Pull request merged**